projects/organization-management/src/app/store/users/users.reducer.ts

```diff
@@ -46,6 +49,16 @@ export function usersReducer(state = initialState, action: UsersAction): UsersSt
         error: undefined,
       };
     }
+
+    case UsersActionTypes.LoadUserRolesSuccess: {
+      const { user, userRoles } = action.payload;
+
+      return {
+        ...usersAdapter.upsertOne({ ...user, roles: userRoles }, state),
```

Comment:
Overwriting the complete user data seems unnecessary. Just updating the roles on the user object would suffice.